File: Problems/0013_roman_integer/main.go
package main

func romanToInt(s string) int {
    
    numbers := map[string]int { 
        "I": 1, 
        "V": 5,
        "X": 10,
        "L": 50,
        "C": 100,
        "D": 500,
        "M": 1000,
    }

    number := 0

    for a := 0 ;  a <= len(s)  ; a++ {
        if  a +1  == len(s) {
            number += numbers[string(s[a])]
            return number
        }

        if numbers[string(s[a])] >= numbers[string(s[a+1])] {
            number += numbers[string(s[a])]
        }   else {
            number -= numbers[string(s[a])]
        }

      
    }
    return number 
}

File: Problems/0013_roman_integer/main.ts

function romanToInt(s: string): number {
    
    const romanToIntMap: Record<string, number> = {
        "I": 1,
        "V": 5,
        "X": 10,
        "L": 50,
        "C": 100,
        "D": 500,
        "M": 1000,
    };

    let number = 0;
    for (let i = 0; i < s.length; i++) {
        if (i + 1 < s.length && romanToIntMap[s[i]] < romanToIntMap[s[i + 1]]) {
            number -= romanToIntMap[s[i]];
        } else {
            number += romanToIntMap[s[i]];
        }
    }

    return number;
}